packages/keyring/src/pair/index.ts

```diff
@@ -124,10 +128,10 @@ export default function createPair ({ toSS58, type }: Setup, { publicKey, secret
       }
     },
     derive: (suri: string, meta?: KeyringPair$Meta): KeyringPair => {
-      assert(secretKey, 'Cannot derive on a locked keypair');
+      assert(!isLocked(secretKey), 'Cannot derive on a locked keypair');
 
       const { path } = keyExtractPath(suri);
-      const derived = keyFromPath({ publicKey, secretKey: secretKey }, path, type);
+      const derived = keyFromPath({ publicKey, secretKey: secretKey! }, path, type);
```

Comment:
```suggestion
      const derived = keyFromPath({ publicKey, secretKey: secretKey }, path, type);
```

Reply:
Sorry about this - the linter does complain, but since the isLocked now does it properly, we are ok and can cleanup.

Reply:
I hope I would know more about Typescript but the linter is still complaining. Let me do some homework :)

Reply:
Grrrr... so that just says the value should not be empty. And the assert/isLocked checks for that - so should just work as expected.

Reply:
The linter doesn't allow `secretKey` without the not-null assertion.

Just read the doc about [type predicates](https://www.typescriptlang.org/docs/handbook/advanced-types.html#using-type-predicates). It looks like a function with a type predicate return type notation is used for narrowing down a type. And the predicates should be used in compiling time. However `isLocked` is used in runtime because it also checks the content of `secretKey`. So it sounds to me a better way is to return a boolean and add change it back to `secretKey: secretKey!`.

Reply:
Something is funny here. But anyway, a warning (although it is only that way since al the `!`'s have not been cleaned - so many hours in a day).

Not going to waste your time, going to get it is since it works.

Reply:
*slush* Was my bug, not yours.

So `isLocked(Uint8Array | undefined)` is true if not a valid (for us), Uint8Array. So effectively we should return "it is not a (valid) Uint8Array".

Adjusting it somewhat in my linting branch.